jwst/pathloss/pathloss.py

```diff
@@ -213,26 +212,31 @@ def do_correction(input_model, pathloss_model):
                                                         aperture.uniform_wcs,
                                                         xcenter, ycenter)
                     if is_inside_slitlet:
-                        #
+
                         # Wavelengths in the reference file are in meters,
                         #need them to be in microns
                         wavelength_pointsource *= 1.0e6
                         wavelength_uniformsource *= 1.0e6
-                        slit.pathloss_pointsource = pathloss_pointsource_vector
-                        slit.wavelength_pointsource =  wavelength_pointsource
-                        slit.pathloss_uniformsource = pathloss_uniform_vector
-                        slit.wavelength_uniformsource = wavelength_uniformsource
-                        #
-                        # Create the 2-d pathloss arrays
+
                         wavelength_array = slit.wavelength
-                        pathloss_pointsource_2d = interpolate_onto_grid(wavelength_array,
-                                                                        wavelength_pointsource,
-                                                                        pathloss_pointsource_vector)
-                        pathloss_uniformsource_2d = interpolate_onto_grid(wavelength_array,
-                                                                          wavelength_uniformsource,
-                                                                          pathloss_uniform_vector)
-                        slit.pathloss_pointsource2d = pathloss_pointsource_2d
-                        slit.pathloss_uniformsource2d = pathloss_uniformsource_2d
+
+                        # Compute and apply the 2D correction
+                        if is_pointsource(slit.source_type):
+                            pathloss_pointsource_2d = interpolate_onto_grid(
+                                wavelength_array,
+                                wavelength_pointsource,
+                                pathloss_pointsource_vector)
+                            slit.data /= pathloss_pointsource_2d
+                            slit.err /= pathloss_pointsource_2d
+                            slit.var_poisson /= pathloss_pointsource_2d**2
+                        else:
+                            pathloss_uniformsource_2d = interpolate_onto_grid(
+                                wavelength_array,
+                                wavelength_uniformsource,
+                                pathloss_uniform_vector)
+                            slit.data /= pathloss_uniformsource_2d
+                            slit.err /= pathloss_uniformsource_2d
+                            slit.var_poisson /= pathloss_uniformsource_2d**2
```

Comment:
Need to save either pathloss_pointsource_2d or pathloss_uniformsource_2d - whichever one was applied - to the slit instance, so it appears in the output product.